cmd/root/collection/cluster.go

```diff
@@ -28,35 +29,39 @@ import (
 
 func ClusterK8sExecute(namespace string, cs CopyStrategy, ddfs helpers.Filesystem, c Collector, k string) error {
 	cmds := []string{"nodes", "sc", "pvc", "pv", "service", "endpoints", "pods", "deployments", "statefulsets", "daemonset", "replicaset", "cronjob", "job", "events", "ingress", "limitrange", "resourcequota", "hpa", "pdb", "pc"}
+	var wg sync.WaitGroup
 	for _, cmd := range cmds {
-		resource := cmd
-		out, err := clusterExecuteBytes(namespace, resource, c, k)
-		if err != nil {
-			simplelog.Errorf("when getting cluster config, error was %v", err)
-			continue
-		}
-		text, err := masking.RemoveSecretsFromK8sJSON(string(out))
-		if err != nil {
-			simplelog.Errorf("unable to mask secrets for %v in namespace %v returning am empty text due to error '%v'", k, namespace, err)
-			continue
-		}
-		p, err := cs.CreatePath("kubernetes", "dremio-master", "")
-		if err != nil {
-			simplelog.Errorf("trying to construct cluster config path %v with error %v", p, err)
-			continue
-		}
-		path := strings.TrimSuffix(p, "dremio-master")
-		filename := filepath.Join(path, resource+".json")
-		err = ddfs.WriteFile(filename, []byte(text), DirPerms)
-		if err != nil {
-			simplelog.Errorf("trying to write file %v, error was %v", filename, err)
-			continue
-		}
+		wg.Add(1)
+		go func(cmdname string) {
+			defer wg.Done()
+			resource := cmdname
+			out, err := clusterExecuteBytes(namespace, resource, c, k)
+			if err != nil {
+				simplelog.Errorf("when getting cluster config, error was %v", err)
```

Comment:
need to return the function here. no need to continue

Reply:
Done